framework/include/problems/EigenProblem.h

```diff
@@ -105,12 +155,37 @@ class EigenProblem : public FEProblemBase
                                  TagID tagA,
                                  TagID tagB);
 
-  void scaleEigenvector(const Real scaling_factor);
+  /**
+   * For nonlinear eigen solver, a good initial value can help convergence.
+   * Should set initial values for only eigen variables.
+   */
+  void initEigenvector(const Real initial_value);
 
   /**
    * Which eigenvalue is active
    */
   unsigned int activeEigenvalueIndex() { return _active_eigen_index; }
```

Comment:
Not in the diff, but let's make this a `const` method

Reply:
Yes